blockchain/cache.go

```diff
@@ -20,20 +20,42 @@ package blockchain
 
 import (
 	"fmt"
-	"github.com/golang/groupcache/lru"
+	"github.com/getsentry/sentry-go"
 	"github.com/google/uuid"
+	lru "github.com/hashicorp/golang-lru"
 )
 
 // newEmulatorCache returns a new instance of cache with provided capacity.
 func newEmulatorCache(capacity int) *emulatorCache {
-	c := lru.New(capacity)
-	c.OnEvicted = func(key lru.Key, value interface{}) {
-		fmt.Printf("Cache evicted emulator for project: %s - (%v)\n", key.(uuid.UUID).String(), key)
+	emCache := &emulatorCache{
+		capacity: capacity,
+		cache:    nil,
+	}
+	_ = emCache.initializeCache()
+	return emCache
+}
+
+// setCache sets emulatorCache to a new lru cache and returns true if successful
+func (c *emulatorCache) initializeCache() bool {
```

Comment:
This feels a bit awkward, not sure if idiomatic in Go, but the way you could go around this if your purpose is to make sure no one initializes cache directly from struct, is to create submodule called cache and then expose only NewEmulatorCache method.